src/llmcompressor/transformers/finetune/session_mixin.py

```diff
@@ -79,15 +79,22 @@ def __init__(
 
         # parse training and metadata args
         training_args = kwargs.get("args")
-        self.metadata = (
-            self._extract_metadata(
+
+        self.metadata = None
+        if training_args is not None:
+            # trl_sft_trainer pathway. Both training_args and data_args
+            # have `max_seq_length` which causes collision error. This is the
+            # only shared parameter, where training arg is `TRLSFTConfig` that
+            # inherits HuggingFace's `TrainingArguments`
+            training_args_dict = training_args.to_dict()
+            if "max_seq_length" in training_args_dict:
+                training_args_dict.pop("max_seq_length")
```

Comment:
nit: `training_args_dict.pop("max_seq_length", None)`

Reply:
Sorry, why are we removing `max_seq_length`? Should we not throw a warning/ replacing the data_args max seq len with the one passed by the user?

Reply:
And these dict is used to populate the metadata, which is used to populate the state. Metadata i think is used to keep history info.